### ISSUE TITLE
Microk8s 1.21 fails to start after controlled cluster shutdown and restart

### ISSUE THREAD

**Issue by crookstonmet** [canonical/microk8s#3114]
(This ticket created based on https://discuss.kubernetes.io/t/microk8s-1-21-fails-to-start-after-controlled-cluster-shutdown-and-restart/19831)

I apologize for the inconvenience, but I can't just attach the tarball here. I'm happy to post individual results as necessary.

I have a 6-node `microk8s` cluster that’s been running pretty well for several months. We were recently notified that power would be cut to our office for 24 hours, so our IT folks shut down the cluster in a controlled manner before bringing it back up after power was restored. Previously, we’ve rebooted individual nodes without issue, but we never had all shut down at the same time.

Since then, we have been unable to fully restart the k8s services – any `kubectl` call is met with:
```
The connection to the server 127.0.0.1:16443 was refused - did you specify the right host or port?
```
I eventually shut down `microk8s` on all the cluster nodes before starting it on a single node. My plan is to get `microk8s` working again on one node before trying to fix them all. When I run `microk8s` status I get `microk8s is not running. Use microk8s inspect for a deeper inspection`. When I run `microk8s inspect` for a deeper inspection I get:
```
Inspecting Certificates
Inspecting services
  Service snap.microk8s.daemon-cluster-agent is running
  Service snap.microk8s.daemon-containerd is running
  Service snap.microk8s.daemon-apiserver-kicker is running
  Service snap.microk8s.daemon-kubelite is running
  Copy service arguments to the final report tarball
Inspecting AppArmor configuration
Gathering system information
  Copy processes list to the final report tarball
  Copy snap list to the final report tarball
  Copy VM name (or none) to the final report tarball
  Copy disk usage information to the final report tarball
  Copy memory usage information to the final report tarball
  Copy server uptime to the final report tarball
  Copy current linux distribution to the final report tarball
  Copy openSSL information to the final report tarball
  Copy network configuration to the final report tarball
Inspecting kubernetes cluster
  Inspect kubernetes cluster
Inspecting juju
  Inspect Juju
Inspecting kubeflow
  Inspect Kubeflow
```

In the discussion referenced above, I posted about how starting a single node was failing. balchua1 suggested that I start at least 3 in case the issue was `dqlite` being unable to form a quorum, and sure enough, starting 3 led to different and hopefully more actionable errors:

On machine 3 it complains:
```
Apr 30 01:26:37 machine-name-3 microk8s.daemon-kubelite[30539]: I0430 01:26:37.909185   30539 server.go:654] external host was not specified, using <IP #3>
Apr 30 01:26:41 machine-name-3 microk8s.daemon-kubelite[30539]: I0430 01:26:41.309369   30539 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read <IP #3>:19001-><IP #2>:38944: read: connection reset by peer
Apr 30 01:26:51 machine-name-3 systemd[1]: snap.microk8s.daemon-kubelite.service: Main process exited, code=killed, status=11/SEGV
```
On machine 2:
```
Apr 30 02:40:01 machine-name-2 microk8s.daemon-kubelite[42482]: I0430 02:40:01.125471   42482 server.go:654] external host was not specified, using <IP #2>
Apr 30 02:40:06 machine-name-2 microk8s.daemon-kubelite[42482]: I0430 02:40:06.145030   42482 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #2>:19001-><IP #1>:41006: read: connection reset by peer
```
And on machine 1:
```
Apr 30 02:42:46 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:42:46.230307   48129 server.go:654] external host was not specified, using <IP #1>
Apr 30 02:42:50 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:42:50.026909   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #2>:42976: read: connection reset by peer
Apr 30 02:43:00 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:43:00.483584   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #3>:58952: read: connection reset by peer
Apr 30 02:43:06 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:43:06.016426   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #2>:43238: read: connection reset by peer
Apr 30 02:43:12 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:43:12.076984   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #2>:43338: read: connection reset by peer
Apr 30 02:43:28 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:43:28.466169   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #3>:59424: read: connection reset by peer
Apr 30 02:43:48 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:43:48.977108   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #2>:43950: read: connection reset by peer
Apr 30 02:44:13 machine-name-1 microk8s.daemon-kubelite[48129]: I0430 02:44:13.007527   48129 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP #1>:19001-><IP #2>:44344: read: connection reset by peer
```

Based on this, I believe the three machines know about each other, but they're unable to replicate their datastores to each other. The only existing ticket I've found that looks similar (https://github.com/canonical/microk8s/issues/1868) had a happy but uninformative resolution. (I wouldn't complain if it happened to me too!) I can confirm that pinging each machine from the other machines succeeds, and `traceroute -p 19001 <machine X>` also works, but those messages still make me think there's a networking issue. Not sure what to check next, though.

I'm attaching 3 log files that show the output of a single attempt to start the service on each of the 3 machines (from which the above snippets were taken). Many thanks for any advice on how to debug this!

[jrnctl_kubelite_1.log](https://github.com/canonical/microk8s/files/8595568/jrnctl_kubelite_1.log)
[jrnctl_kubelite_2.log](https://github.com/canonical/microk8s/files/8595569/jrnctl_kubelite_2.log)
[jrnctl_kubelite_3.log](https://github.com/canonical/microk8s/files/8595570/jrnctl_kubelite_3.log)

**Comment by balchua** [comment]
Can you also print out the content of `/var/snap/microk8s/current/var/kubernetes/backend/cluster.yaml `, and try starting those nodes with `Role: 0`,  Nodes with `Role: 0` where the active nodes prior to it being shutdown.

Before you gather the logs let it sit for a while, i think it will try to join and see whether they can form the cluster.  :crossed_fingers:

**Comment by crookstonmet** [comment]
The `cluster.yaml` files are identical on all three -- the three started nodes have role 0:
```
- Address: <IP 4>:19001
  ID: 3297041220608546238
  Role: 1
- Address: <IP 3>:19001
  ID: 4627570732265678721
  Role: 0
- Address: <IP 1>:19001
  ID: 6160109552035312981
  Role: 0
- Address: <IP 2>:19001
  ID: 13403531148419318862
  Role: 0
- Address: <IP 6>:19001
  ID: 8060500096391682189
  Role: 1
- Address: <IP 5>:19001
  ID: 3325712208165750247
  Role: 1
```

Looking at the logs around the same time:
Node 1:
```
Apr 30 14:33:35 machine-1 microk8s.daemon-kubelite[52422]: I0430 14:33:35.241944   52422 daemon.go:65] Starting API Server
Apr 30 14:33:35 machine-1 microk8s.daemon-kubelite[52422]: Flag --insecure-port has been deprecated, This flag has no effect now and will be removed in v1.24.
Apr 30 14:33:35 machine-1 microk8s.daemon-kubelite[52422]: I0430 14:33:35.243576   52422 server.go:654] external host was not specified, using <IP 1>
Apr 30 14:33:38 machine-1 microk8s.daemon-kubelite[52422]: I0430 14:33:38.462912   52422 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP 1>:19001-><IP 3>:35472: read: connection reset by peer
Apr 30 14:33:48 machine-1 microk8s.daemon-kubelite[52422]: I0430 14:33:48.973905   52422 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP 1>:19001-><IP 3>:35664: read: connection reset by peer
Apr 30 14:33:59 machine-1 systemd[1]: snap.microk8s.daemon-kubelite.service: Main process exited, code=killed, status=11/SEGV
Apr 30 14:33:59 machine-1 systemd[1]: snap.microk8s.daemon-kubelite.service: Failed with result 'signal'.
```
Node 2:
```
Apr 30 14:33:31 machine-2 microk8s.daemon-kubelite[11992]: I0430 14:33:31.622704   11992 daemon.go:65] Starting API Server
Apr 30 14:33:31 machine-2 microk8s.daemon-kubelite[11992]: Flag --insecure-port has been deprecated, This flag has no effect now and will be removed in v1.24.
Apr 30 14:33:31 machine-2 microk8s.daemon-kubelite[11992]: I0430 14:33:31.626180   11992 server.go:654] external host was not specified, using <IP 2>
Apr 30 14:33:34 machine-2 microk8s.daemon-kubelite[11992]: I0430 14:33:34.832811   11992 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP 2>:19001-><IP 1>:45372: read: connection reset by peer
Apr 30 14:33:42 machine-2 microk8s.daemon-kubelite[11992]: I0430 14:33:42.696481   11992 log.go:184] [ERROR] dqlite: proxy: first: remote -> local: read tcp <IP 2>:19001-><IP 3>:57380: read: connection reset by peer
Apr 30 14:34:03 machine-2 systemd[1]: snap.microk8s.daemon-kubelite.service: Main process exited, code=killed, status=11/SEGV
Apr 30 14:34:03 machine-2 systemd[1]: snap.microk8s.daemon-kubelite.service: Failed with result 'signal'.
```
Node 3:
```
Apr 30 14:33:25 machine-3 microk8s.daemon-kubelite[59897]: I0430 14:33:25.655029   59897 daemon.go:65] Starting API Server
Apr 30 14:33:25 machine-3 microk8s.daemon-kubelite[59897]: Flag --insecure-port has been deprecated, This flag has no effect now and will be removed in v1.24.
Apr 30 14:33:25 machine-3 microk8s.daemon-kubelite[59897]: I0430 14:33:25.659780   59897 server.go:654] external host was not specified, using <IP 3>
Apr 30 14:33:38 machine-3 systemd[1]: snap.microk8s.daemon-kubelite.service: Main process exited, code=killed, status=11/SEGV
Apr 30 14:33:38 machine-3 systemd[1]: snap.microk8s.daemon-kubelite.service: Failed with result 'signal'.
```
In other words, Node 3 doesn't report failing to connect to anyone, Node 1 attempts to contact Node 3, and Node 2 attempts to contact them both. The time between starting the API server and the service being killed is just a few seconds if it doesn't report contacting anyone, and not much longer (~30 seconds) if it does. However, on other iterations Node 3 does attempt to contact the other nodes. It seems like the order is random, depending on when each service was last killed. It seems like Node 3, which started first in that iteration, isn't responding to connections from the other two before it dies. Can you think of a way to see why Node 3 (in this case) is dying? I believe 11 is a catch-all signal. 

Logs of the last thousand entries for each node are attached. Thank you for your help!

[jrnctl_kubelite_1.log](https://github.com/canonical/microk8s/files/8597138/jrnctl_kubelite_1.log)
[jrnctl_kubelite_2.log](https://github.com/canonical/microk8s/files/8597139/jrnctl_kubelite_2.log)
[jrnctl_kubelite_3.log](https://github.com/canonical/microk8s/files/8597140/jrnctl_kubelite_3.log)

**Comment by balchua** [comment]
Can you show the result of `snap info microk8s`?  status `11/SEGV` is more or less a result of dqlite failing to start, resulting to segmentation fault.  In the newer release of MicroK8s and dqlite, they added environment variables to print debug logs.  But I am not sure if this is in your version.

Can you also provide the content of `/var/snap/microk8s/current/var/kubernetes/backend` in case the MicroK8s team can find it useful.
@neoaggelos @ktsakalozos , if you guys have any idea where to look. :) I am stumped with this issue

**Comment by crookstonmet** [comment]
The info for Node 2 is the same as the other two:
```
$ snap info microk8s
name:      microk8s
summary:   Kubernetes for workstations and appliances
publisher: Canonical✓
store-url: https://snapcraft.io/microk8s
contact:   https://github.com/ubuntu/microk8s
license:   unset
description: |
  MicroK8s is a small, fast, secure, single node Kubernetes that installs on
  just about any Linux box. Use it for offline development, prototyping,
  testing, or use it on a VM as a small, cheap, reliable k8s for CI/CD. It's
  also a great k8s for appliances - develop your IoT apps for k8s and deploy
  them to MicroK8s on your boxes.
commands:
  - microk8s.add-node
  - microk8s.cilium
  - microk8s.config
  - microk8s.ctr
  - microk8s.dashboard-proxy
  - microk8s.dbctl
  - microk8s.disable
  - microk8s.enable
  - microk8s.helm
  - microk8s.helm3
  - microk8s.inspect
  - microk8s.istioctl
  - microk8s.join
  - microk8s.juju
  - microk8s.kubectl
  - microk8s.leave
  - microk8s.linkerd
  - microk8s
  - microk8s.refresh-certs
  - microk8s.remove-node
  - microk8s.reset
  - microk8s.start
  - microk8s.status
  - microk8s.stop
services:
  microk8s.daemon-apiserver:            simple, enabled, inactive
  microk8s.daemon-apiserver-kicker:     simple, enabled, active
  microk8s.daemon-cluster-agent:        simple, enabled, active
  microk8s.daemon-containerd:           simple, enabled, active
  microk8s.daemon-control-plane-kicker: simple, enabled, inactive
  microk8s.daemon-controller-manager:   simple, enabled, inactive
  microk8s.daemon-etcd:                 simple, enabled, inactive
  microk8s.daemon-flanneld:             simple, enabled, inactive
  microk8s.daemon-kubelet:              simple, enabled, inactive
  microk8s.daemon-kubelite:             simple, enabled, active
  microk8s.daemon-proxy:                simple, enabled, inactive
  microk8s.daemon-scheduler:            simple, enabled, inactive
snap-id:      EaXqgt1lyCaxKaQCU349mlodBkDCXRcg
tracking:     1.21/stable
refresh-date: 33 days ago, at 17:10 UTC
channels:
  1.23/stable:      v1.23.6         2022-04-29 (3204) 218MB classic
  1.23/candidate:   v1.23.6         2022-04-28 (3204) 218MB classic
  1.23/beta:        v1.23.6         2022-04-28 (3204) 218MB classic
  1.23/edge:        v1.23.6         2022-04-28 (3204) 218MB classic
  latest/stable:    v1.23.5         2022-03-30 (3052) 219MB classic
  latest/candidate: v1.23.6         2022-04-29 (3206) 230MB classic
  latest/beta:      v1.23.6         2022-04-29 (3206) 230MB classic
  latest/edge:      v1.23.6         2022-04-29 (3213) 230MB classic
  dqlite/stable:    –                                       
  dqlite/candidate: –                                       
  dqlite/beta:      –                                       
  dqlite/edge:      v1.16.2         2019-11-07 (1038) 189MB classic
  1.24/stable:      –                                       
  1.24/candidate:   v1.24.0-rc.1    2022-04-27 (3194) 230MB classic
  1.24/beta:        v1.24.0-beta.0  2022-04-01 (3089) 223MB classic
  1.24/edge:        v1.24.0-alpha.4 2022-03-23 (3068) 223MB classic
  1.22/stable:      v1.22.8         2022-03-29 (3057) 193MB classic
  1.22/candidate:   v1.22.9         2022-04-28 (3203) 193MB classic
  1.22/beta:        v1.22.9         2022-04-28 (3203) 193MB classic
  1.22/edge:        v1.22.9         2022-04-28 (3203) 193MB classic
  1.21/stable:      v1.21.11        2022-03-29 (3058) 191MB classic
  1.21/candidate:   v1.21.12        2022-04-29 (3202) 191MB classic
  1.21/beta:        v1.21.12        2022-04-29 (3202) 191MB classic
  1.21/edge:        v1.21.12        2022-04-28 (3202) 191MB classic
  1.20/stable:      v1.20.13        2021-12-08 (2760) 221MB classic
  1.20/candidate:   v1.20.13        2021-12-07 (2760) 221MB classic
  1.20/beta:        v1.20.13        2021-12-07 (2760) 221MB classic
  1.20/edge:        v1.20.14        2022-01-11 (2843) 217MB classic
  1.19/stable:      v1.19.15        2021-09-30 (2530) 216MB classic
  1.19/candidate:   v1.19.15        2021-09-29 (2530) 216MB classic
  1.19/beta:        v1.19.15        2021-09-29 (2530) 216MB classic
  1.19/edge:        v1.19.16        2022-01-07 (2820) 212MB classic
  1.18/stable:      v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/candidate:   v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/beta:        v1.18.20        2021-07-12 (2271) 198MB classic
  1.18/edge:        v1.18.20        2021-11-03 (2647) 198MB classic
  1.17/stable:      v1.17.17        2021-01-15 (1916) 177MB classic
  1.17/candidate:   v1.17.17        2021-01-14 (1916) 177MB classic
  1.17/beta:        v1.17.17        2021-01-14 (1916) 177MB classic
  1.17/edge:        v1.17.17        2021-01-13 (1916) 177MB classic
  1.16/stable:      v1.16.15        2020-09-12 (1671) 179MB classic
  1.16/candidate:   v1.16.15        2020-09-04 (1671) 179MB classic
  1.16/beta:        v1.16.15        2020-09-04 (1671) 179MB classic
  1.16/edge:        v1.16.15        2020-09-02 (1671) 179MB classic
  1.15/stable:      v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/candidate:   v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/beta:        v1.15.11        2020-03-27 (1301) 171MB classic
  1.15/edge:        v1.15.11        2020-03-26 (1301) 171MB classic
  1.14/stable:      v1.14.10        2020-01-06 (1120) 217MB classic
  1.14/candidate:   ↑                                       
  1.14/beta:        ↑                                       
  1.14/edge:        v1.14.10        2020-03-26 (1303) 217MB classic
  1.13/stable:      v1.13.6         2019-06-06  (581) 237MB classic
  1.13/candidate:   ↑                                       
  1.13/beta:        ↑                                       
  1.13/edge:        ↑                                       
  1.12/stable:      v1.12.9         2019-06-06  (612) 259MB classic
  1.12/candidate:   ↑                                       
  1.12/beta:        ↑                                       
  1.12/edge:        ↑                                       
  1.11/stable:      v1.11.10        2019-05-10  (557) 258MB classic
  1.11/candidate:   ↑                                       
  1.11/beta:        ↑                                       
  1.11/edge:        ↑                                       
  1.10/stable:      v1.10.13        2019-04-22  (546) 222MB classic
  1.10/candidate:   ↑                                       
  1.10/beta:        ↑                                       
  1.10/edge:        ↑                                       
installed:          v1.21.11                   (3058) 191MB classic
```
And the listing of Node 2's `backend` folder (which is again representative of all the nodes):
```
-rw-rw---- 1 root microk8s   7570200 Apr 24 23:28 0000000404287381-0000000404288130
-rw-rw---- 1 root microk8s   4227056 Apr 24 23:32 0000000404288131-0000000404288611
-rw-rw---- 1 root microk8s   8374384 Apr 24 23:38 0000000404288612-0000000404289066
-rw-rw---- 1 root microk8s   1307312 Apr 24 23:39 0000000404289067-0000000404289170
-rw-rw---- 1 root microk8s   5721488 Apr 24 23:46 0000000404289171-0000000404289760
-rw-rw---- 1 root microk8s   5879920 Apr 24 23:52 0000000404289761-0000000404290251
-rw-rw---- 1 root microk8s   4730624 Apr 24 23:58 0000000404290252-0000000404290727
-rw-rw---- 1 root microk8s   8339384 Apr 25 00:06 0000000404290728-0000000404291587
-rw-rw---- 1 root microk8s    477440 Apr 25 00:06 0000000404291588-0000000404291606
-rw-rw---- 1 root microk8s   6589160 Apr 25 00:12 0000000404291607-0000000404292151
-rw-rw---- 1 root microk8s   6981536 Apr 25 00:20 0000000404292152-0000000404293162
-rw-rw---- 1 root microk8s  14057776 Apr 25 00:34 0000000404293163-0000000404293933
-rw-rw---- 1 root microk8s   2786664 Apr 25 00:38 0000000404293934-0000000404294170
-rw-rw---- 1 root microk8s   8383376 Apr 25 00:45 0000000404294171-0000000404295115
-rw-rw---- 1 root microk8s    337616 Apr 25 00:46 0000000404295116-0000000404295130
-rw-rw---- 1 root microk8s   5122120 Apr 25 00:54 0000000404295131-0000000404295634
-rw-rw---- 1 root microk8s   4646152 Apr 25 00:57 0000000404295635-0000000404296269
-rw-rw---- 1 root microk8s   7961248 Apr 25 00:57 0000000404296270-0000000404297413
-rw-rw---- 1 root microk8s        48 Apr 25 02:40 0000000404297414-0000000404297414
-rw-rw---- 1 root microk8s       304 Apr 25 02:41 0000000404297415-0000000404297417
-rw-rw---- 1 root microk8s        88 Apr 25 02:42 0000000404297418-0000000404297419
-rw-rw---- 1 root microk8s       176 Apr 25 02:43 0000000404297420-0000000404297424
-rw-rw---- 1 root microk8s        72 Apr 25 02:43 0000000404297425-0000000404297426
-rw-rw---- 1 root microk8s        88 Apr 25 02:44 0000000404297427-0000000404297428
-rw-rw---- 1 root microk8s        48 Apr 25 02:44 0000000404297429-0000000404297429
-rw-rw---- 1 root microk8s        48 Apr 25 02:44 0000000404297430-0000000404297430
-rw-rw---- 1 root microk8s        48 Apr 25 02:45 0000000404297431-0000000404297431
-rw-rw---- 1 root microk8s       112 Apr 25 02:46 0000000404297432-0000000404297434
-rw-rw---- 1 root microk8s        88 Apr 25 02:47 0000000404297435-0000000404297436
-rw-rw---- 1 root microk8s        72 Apr 25 02:47 0000000404297437-0000000404297438
-rw-rw---- 1 root microk8s        48 Apr 25 02:48 0000000404297439-0000000404297439
-rw-rw---- 1 root microk8s        48 Apr 25 02:48 0000000404297440-0000000404297440
-rw-rw---- 1 root microk8s        96 Apr 25 02:49 0000000404297441-0000000404297443
-rw-rw---- 1 root microk8s       128 Apr 25 02:50 0000000404297444-0000000404297446
-rw-rw---- 1 root microk8s        48 Apr 25 02:50 0000000404297447-0000000404297447
-rw-rw---- 1 root microk8s        88 Apr 25 02:50 0000000404297448-0000000404297449
-rw-rw---- 1 root microk8s        72 Apr 25 02:51 0000000404297450-0000000404297451
-rw-rw---- 1 root microk8s        72 Apr 25 02:51 0000000404297452-0000000404297453
-rw-rw---- 1 root microk8s        48 Apr 25 02:52 0000000404297454-0000000404297454
-rw-rw---- 1 root microk8s       200 Apr 25 02:52 0000000404297455-0000000404297460
-rw-rw---- 1 root microk8s     29072 Apr 25 02:54 0000000404297461-0000000404297472
-rw-rw---- 1 root microk8s       240 Apr 25 02:54 0000000404297473-0000000404297481
-rw-rw---- 1 root microk8s        48 Apr 25 02:54 0000000404297482-0000000404297482
-rw-rw---- 1 root microk8s       144 Apr 25 02:54 0000000404297483-0000000404297487
-rw-rw---- 1 root microk8s       240 Apr 25 02:54 0000000404297488-0000000404297496
-rw-rw---- 1 root microk8s       384 Apr 25 02:55 0000000404297497-0000000404297511
-rw-rw---- 1 root microk8s       328 Apr 25 02:55 0000000404297512-0000000404297519
-rw-rw---- 1 root microk8s       480 Apr 25 02:55 0000000404297520-0000000404297538
-rw-rw---- 1 root microk8s       336 Apr 25 02:56 0000000404297539-0000000404297551
-rw-rw---- 1 root microk8s        48 Apr 25 03:38 0000000404297552-0000000404297552
-rw-rw---- 1 root microk8s       112 Apr 25 03:38 0000000404297553-0000000404297555
-rw-rw---- 1 root microk8s        48 Apr 25 03:56 0000000404297556-0000000404297556
-rw-rw---- 1 root microk8s        88 Apr 25 03:56 0000000404297557-0000000404297558
-rw-rw---- 1 root microk8s       312 Apr 25 03:59 0000000404297559-0000000404297570
-rw-rw---- 1 root microk8s        96 Apr 25 04:00 0000000404297571-0000000404297573
-rw-rw---- 1 root microk8s        48 Apr 25 04:15 0000000404297574-0000000404297574
-rw-rw---- 1 root microk8s        88 Apr 25 04:15 0000000404297575-0000000404297576
-rw-rw---- 1 root microk8s        48 Apr 25 04:23 0000000404297577-0000000404297577
-rw-rw---- 1 root microk8s        48 Apr 25 04:23 0000000404297578-0000000404297578
-rw-rw---- 1 root microk8s        96 Apr 25 04:35 0000000404297579-0000000404297581
-rw-rw---- 1 root microk8s        72 Apr 25 05:52 0000000404297582-0000000404297583
-rw-rw---- 1 root microk8s        48 Apr 25 06:09 0000000404297584-0000000404297584
-rw-rw---- 1 root microk8s        48 Apr 25 06:10 0000000404297585-0000000404297585
-rw-rw---- 1 root microk8s        96 Apr 25 06:16 0000000404297586-0000000404297588
-rw-rw---- 1 root microk8s        88 Apr 25 06:35 0000000404297589-0000000404297590
-rw-rw---- 1 root microk8s       112 Apr 25 06:35 0000000404297591-0000000404297593
-rw-rw---- 1 root microk8s       112 Apr 25 06:35 0000000404297594-0000000404297596
-rw-rw---- 1 root microk8s        48 Apr 25 06:35 0000000404297597-0000000404297597
-rw-rw---- 1 root microk8s        48 Apr 25 06:36 0000000404297598-0000000404297598
-rw-rw---- 1 root microk8s       128 Apr 25 06:36 0000000404297599-0000000404297601
-rw-rw---- 1 root microk8s        48 Apr 25 06:36 0000000404297602-0000000404297602
-rw-rw---- 1 root microk8s     16664 Apr 25 06:38 0000000404297603-0000000404297610
-rw-rw---- 1 root microk8s       144 Apr 25 06:38 0000000404297611-0000000404297615
-rw-rw---- 1 root microk8s       336 Apr 25 06:39 0000000404297616-0000000404297628
-rw-rw---- 1 root microk8s        48 Apr 25 07:05 0000000404297629-0000000404297629
-rw-rw---- 1 root microk8s       128 Apr 25 07:05 0000000404297630-0000000404297632
-rw-rw---- 1 root microk8s       120 Apr 25 07:36 0000000404297633-0000000404297636
-rw-rw---- 1 root microk8s        48 Apr 25 07:50 0000000404297637-0000000404297637
-rw-rw---- 1 root microk8s       128 Apr 25 07:50 0000000404297638-0000000404297640
-rw-rw---- 1 root microk8s        72 Apr 25 07:50 0000000404297641-0000000404297642
-rw-rw---- 1 root microk8s        48 Apr 25 07:51 0000000404297643-0000000404297643
-rw-rw---- 1 root microk8s       128 Apr 25 07:51 0000000404297644-0000000404297646
-rw-rw---- 1 root microk8s       168 Apr 25 07:57 0000000404297647-0000000404297652
-rw-rw---- 1 root microk8s        48 Apr 25 08:34 0000000404297653-0000000404297653
-rw-rw---- 1 root microk8s        88 Apr 25 08:34 0000000404297654-0000000404297655
-rw-rw---- 1 root microk8s        48 Apr 25 08:55 0000000404297656-0000000404297656
-rw-rw---- 1 root microk8s        88 Apr 25 08:55 0000000404297657-0000000404297658
-rw-rw---- 1 root microk8s        48 Apr 25 09:40 0000000404297659-0000000404297659
-rw-rw---- 1 root microk8s        88 Apr 25 09:41 0000000404297660-0000000404297661
-rw-rw---- 1 root microk8s       144 Apr 25 09:46 0000000404297662-0000000404297666
-rw-rw---- 1 root microk8s        72 Apr 25 09:55 0000000404297667-0000000404297668
-rw-rw---- 1 root microk8s        48 Apr 25 10:04 0000000404297669-0000000404297669
-rw-rw---- 1 root microk8s        48 Apr 25 10:05 0000000404297670-0000000404297670
-rw-rw---- 1 root microk8s        48 Apr 25 10:09 0000000404297671-0000000404297671
-rw-rw---- 1 root microk8s        48 Apr 25 10:10 0000000404297672-0000000404297672
-rw-rw---- 1 root microk8s        48 Apr 25 10:17 0000000404297673-0000000404297673
-rw-rw---- 1 root microk8s        72 Apr 25 10:18 0000000404297674-0000000404297675
-rw-rw---- 1 root microk8s        48 Apr 25 10:18 0000000404297676-0000000404297676
-rw-rw---- 1 root microk8s        96 Apr 25 10:38 0000000404297677-0000000404297679
-rw-rw---- 1 root microk8s       120 Apr 25 10:49 0000000404297680-0000000404297683
-rw-rw---- 1 root microk8s        48 Apr 25 10:51 0000000404297684-0000000404297684
-rw-rw---- 1 root microk8s        88 Apr 25 10:51 0000000404297685-0000000404297686
-rw-rw---- 1 root microk8s        48 Apr 25 10:56 0000000404297687-0000000404297687
-rw-rw---- 1 root microk8s        72 Apr 25 10:57 0000000404297688-0000000404297689
-rw-rw---- 1 root microk8s        48 Apr 25 11:58 0000000404297690-0000000404297690
-rw-rw---- 1 root microk8s        48 Apr 25 11:58 0000000404297691-0000000404297691
-rw-rw---- 1 root microk8s        48 Apr 25 12:17 0000000404297692-0000000404297692
-rw-rw---- 1 root microk8s        88 Apr 25 12:17 0000000404297693-0000000404297694
-rw-rw---- 1 root microk8s       144 Apr 25 12:29 0000000404297695-0000000404297699
-rw-rw---- 1 root microk8s        48 Apr 25 12:53 0000000404297700-0000000404297700
-rw-rw---- 1 root microk8s       128 Apr 25 12:53 0000000404297701-0000000404297703
-rw-rw---- 1 root microk8s        48 Apr 25 12:59 0000000404297704-0000000404297704
-rw-rw---- 1 root microk8s        48 Apr 25 13:00 0000000404297705-0000000404297705
-rw-rw---- 1 root microk8s        48 Apr 25 13:33 0000000404297706-0000000404297706
-rw-rw---- 1 root microk8s        72 Apr 25 13:34 0000000404297707-0000000404297708
-rw-rw---- 1 root microk8s       120 Apr 25 14:10 0000000404297709-0000000404297712
-rw-rw---- 1 root microk8s        48 Apr 25 14:24 0000000404297713-0000000404297713
-rw-rw---- 1 root microk8s        88 Apr 25 14:24 0000000404297714-0000000404297715
-rw-rw---- 1 root microk8s        48 Apr 25 14:25 0000000404297716-0000000404297716
-rw-rw---- 1 root microk8s       120 Apr 25 15:01 0000000404297717-0000000404297720
-rw-rw---- 1 root microk8s       264 Apr 25 15:22 0000000404297721-0000000404297730
-rw-rw---- 1 root microk8s       456 Apr 25 16:00 0000000404297731-0000000404297748
-rw-rw---- 1 root microk8s        48 Apr 25 16:02 0000000404297749-0000000404297749
-rw-rw---- 1 root microk8s       128 Apr 25 16:02 0000000404297750-0000000404297752
-rw-rw---- 1 root microk8s        96 Apr 25 16:31 0000000404297753-0000000404297755
-rw-rw---- 1 root microk8s       312 Apr 25 17:07 0000000404297756-0000000404297767
-rw-rw---- 1 root microk8s        96 Apr 25 17:18 0000000404297768-0000000404297770
-rw-rw---- 1 root microk8s       312 Apr 25 17:31 0000000404297771-0000000404297782
-rw-rw---- 1 root microk8s        48 Apr 25 17:48 0000000404297783-0000000404297783
-rw-rw---- 1 root microk8s        48 Apr 25 17:48 0000000404297784-0000000404297784
-rw-rw---- 1 root microk8s        48 Apr 25 18:17 0000000404297785-0000000404297785
-rw-rw---- 1 root microk8s        48 Apr 25 18:17 0000000404297786-0000000404297786
-rw-rw---- 1 root microk8s        96 Apr 25 18:24 0000000404297787-0000000404297789
-rw-rw---- 1 root microk8s        96 Apr 25 18:39 0000000404297790-0000000404297792
-rw-rw---- 1 root microk8s        72 Apr 25 18:47 0000000404297793-0000000404297794
-rw-rw---- 1 root microk8s       120 Apr 25 19:00 0000000404297795-0000000404297798
-rw-rw---- 1 root microk8s        48 Apr 25 19:24 0000000404297799-0000000404297799
-rw-rw---- 1 root microk8s       128 Apr 25 19:25 0000000404297800-0000000404297802
-rw-rw---- 1 root microk8s        48 Apr 25 19:27 0000000404297803-0000000404297803
-rw-rw---- 1 root microk8s       168 Apr 25 19:27 0000000404297804-0000000404297807
-rw-rw---- 1 root microk8s        48 Apr 25 20:33 0000000404297808-0000000404297808
-rw-rw---- 1 root microk8s        88 Apr 25 20:33 0000000404297809-0000000404297810
-rw-rw---- 1 root microk8s       168 Apr 25 21:02 0000000404297811-0000000404297816
-rw-rw---- 1 root microk8s        48 Apr 25 21:15 0000000404297817-0000000404297817
-rw-rw---- 1 root microk8s        88 Apr 25 21:16 0000000404297818-0000000404297819
-rw-rw---- 1 root microk8s        72 Apr 25 21:36 0000000404297820-0000000404297821
-rw-rw---- 1 root microk8s       264 Apr 25 21:54 0000000404297822-0000000404297831
-rw-rw---- 1 root microk8s        48 Apr 25 22:23 0000000404297832-0000000404297832
-rw-rw---- 1 root microk8s        48 Apr 25 22:23 0000000404297833-0000000404297833
-rw-rw---- 1 root microk8s       120 Apr 25 22:31 0000000404297834-0000000404297837
-rw-rw---- 1 root microk8s        88 Apr 25 22:45 0000000404297838-0000000404297839
-rw-rw---- 1 root microk8s       240 Apr 25 22:47 0000000404297840-0000000404297848
-rw-rw---- 1 root microk8s        48 Apr 25 22:48 0000000404297849-0000000404297849
-rw-rw---- 1 root microk8s       120 Apr 25 22:54 0000000404297850-0000000404297853
-rw-rw---- 1 root microk8s        72 Apr 25 23:47 0000000404297854-0000000404297855
-rw-rw---- 1 root microk8s        48 Apr 25 23:55 0000000404297856-0000000404297856
-rw-rw---- 1 root microk8s        48 Apr 25 23:55 0000000404297857-0000000404297857
-rw-rw---- 1 root microk8s        48 Apr 26 00:01 0000000404297858-0000000404297858
-rw-rw---- 1 root microk8s        88 Apr 26 00:01 0000000404297859-0000000404297860
-rw-rw---- 1 root microk8s        48 Apr 26 00:20 0000000404297861-0000000404297861
-rw-rw---- 1 root microk8s       128 Apr 26 00:20 0000000404297862-0000000404297864
-rw-rw---- 1 root microk8s        48 Apr 26 02:01 0000000404297865-0000000404297865
-rw-rw---- 1 root microk8s        96 Apr 26 02:02 0000000404297866-0000000404297868
-rw-rw---- 1 root microk8s       144 Apr 26 02:36 0000000404297869-0000000404297873
-rw-rw---- 1 root microk8s        48 Apr 26 03:13 0000000404297874-0000000404297874
-rw-rw---- 1 root microk8s        48 Apr 26 03:14 0000000404297875-0000000404297875
-rw-rw---- 1 root microk8s       384 Apr 26 03:20 0000000404297876-0000000404297890
-rw-rw---- 1 root microk8s        72 Apr 26 03:28 0000000404297891-0000000404297892
-rw-rw---- 1 root microk8s        48 Apr 26 03:35 0000000404297893-0000000404297893
-rw-rw---- 1 root microk8s        88 Apr 26 03:35 0000000404297894-0000000404297895
-rw-rw---- 1 root microk8s       192 Apr 26 03:41 0000000404297896-0000000404297902
-rw-rw---- 1 root microk8s       312 Apr 26 04:06 0000000404297903-0000000404297914
-rw-rw---- 1 root microk8s        48 Apr 26 04:07 0000000404297915-0000000404297915
-rw-rw---- 1 root microk8s       192 Apr 26 04:08 0000000404297916-0000000404297922
-rw-rw---- 1 root microk8s       120 Apr 26 04:40 0000000404297923-0000000404297926
-rw-rw---- 1 root microk8s        48 Apr 26 04:40 0000000404297927-0000000404297927
-rw-rw---- 1 root microk8s       120 Apr 26 04:43 0000000404297928-0000000404297931
-rw-rw---- 1 root microk8s        48 Apr 26 05:26 0000000404297932-0000000404297932
-rw-rw---- 1 root microk8s        96 Apr 26 05:27 0000000404297933-0000000404297935
-rw-rw---- 1 root microk8s        48 Apr 26 05:51 0000000404297936-0000000404297936
-rw-rw---- 1 root microk8s        88 Apr 26 05:51 0000000404297937-0000000404297938
-rw-rw---- 1 root microk8s        96 Apr 26 06:02 0000000404297939-0000000404297941
-rw-rw---- 1 root microk8s        96 Apr 26 06:21 0000000404297942-0000000404297944
-rw-rw---- 1 root microk8s        48 Apr 26 06:25 0000000404297945-0000000404297945
-rw-rw---- 1 root microk8s        48 Apr 26 06:25 0000000404297946-0000000404297946
-rw-rw---- 1 root microk8s        48 Apr 26 06:42 0000000404297947-0000000404297947
-rw-rw---- 1 root microk8s        48 Apr 26 06:42 0000000404297948-0000000404297948
-rw-rw---- 1 root microk8s        48 Apr 26 07:38 0000000404297949-0000000404297949
-rw-rw---- 1 root microk8s        88 Apr 26 07:40 0000000404297950-0000000404297951
-rw-rw---- 1 root microk8s        48 Apr 26 07:41 0000000404297952-0000000404297952
-rw-rw---- 1 root microk8s        48 Apr 26 07:54 0000000404297953-0000000404297953
-rw-rw---- 1 root microk8s        72 Apr 26 07:56 0000000404297954-0000000404297955
-rw-rw---- 1 root microk8s        48 Apr 26 07:56 0000000404297956-0000000404297956
-rw-rw---- 1 root microk8s        88 Apr 26 07:56 0000000404297957-0000000404297958
-rw-rw---- 1 root microk8s        48 Apr 26 07:56 0000000404297959-0000000404297959
-rw-rw---- 1 root microk8s        88 Apr 26 07:57 0000000404297960-0000000404297961
-rw-rw---- 1 root microk8s        88 Apr 26 07:58 0000000404297962-0000000404297963
-rw-rw---- 1 root microk8s        48 Apr 26 07:59 0000000404297964-0000000404297964
-rw-rw---- 1 root microk8s        88 Apr 26 07:59 0000000404297965-0000000404297966
-rw-rw---- 1 root microk8s        88 Apr 26 07:59 0000000404297967-0000000404297968
-rw-rw---- 1 root microk8s        48 Apr 26 07:59 0000000404297969-0000000404297969
-rw-rw---- 1 root microk8s        88 Apr 26 08:01 0000000404297970-0000000404297971
-rw-rw---- 1 root microk8s       128 Apr 26 08:01 0000000404297972-0000000404297974
-rw-rw---- 1 root microk8s       192 Apr 26 08:23 0000000404297975-0000000404297981
-rw-rw---- 1 root microk8s        48 Apr 26 09:17 0000000404297982-0000000404297982
-rw-rw---- 1 root microk8s        48 Apr 26 09:17 0000000404297983-0000000404297983
-rw-rw---- 1 root microk8s       152 Apr 26 09:17 0000000404297984-0000000404297987
-rw-rw---- 1 root microk8s        48 Apr 26 09:18 0000000404297988-0000000404297988
-rw-rw---- 1 root microk8s        48 Apr 26 09:19 0000000404297989-0000000404297989
-rw-rw---- 1 root microk8s        88 Apr 26 09:21 0000000404297990-0000000404297991
-rw-rw---- 1 root microk8s        88 Apr 26 09:21 0000000404297992-0000000404297993
-rw-rw---- 1 root microk8s       128 Apr 26 09:21 0000000404297994-0000000404297996
-rw-rw---- 1 root microk8s        48 Apr 26 09:52 0000000404297997-0000000404297997
-rw-rw---- 1 root microk8s        88 Apr 26 09:52 0000000404297998-0000000404297999
-rw-rw---- 1 root microk8s        48 Apr 26 10:04 0000000404298000-0000000404298000
-rw-rw---- 1 root microk8s        96 Apr 26 10:04 0000000404298001-0000000404298003
-rw-rw---- 1 root microk8s       240 Apr 26 10:27 0000000404298004-0000000404298012
-rw-rw---- 1 root microk8s        48 Apr 26 10:54 0000000404298013-0000000404298013
-rw-rw---- 1 root microk8s       128 Apr 26 10:54 0000000404298014-0000000404298016
-rw-rw---- 1 root microk8s        72 Apr 26 11:08 0000000404298017-0000000404298018
-rw-rw---- 1 root microk8s        48 Apr 26 11:11 0000000404298019-0000000404298019
-rw-rw---- 1 root microk8s       112 Apr 26 11:12 0000000404298020-0000000404298022
-rw-rw---- 1 root microk8s        48 Apr 26 11:12 0000000404298023-0000000404298023
-rw-rw---- 1 root microk8s        72 Apr 26 11:31 0000000404298024-0000000404298025
-rw-rw---- 1 root microk8s        96 Apr 26 11:41 0000000404298026-0000000404298028
-rw-rw---- 1 root microk8s        48 Apr 26 11:48 0000000404298029-0000000404298029
-rw-rw---- 1 root microk8s        88 Apr 26 11:48 0000000404298030-0000000404298031
-rw-rw---- 1 root microk8s        88 Apr 26 11:57 0000000404298032-0000000404298033
-rw-rw---- 1 root microk8s        48 Apr 26 11:57 0000000404298034-0000000404298034
-rw-rw---- 1 root microk8s       128 Apr 26 11:57 0000000404298035-0000000404298037
-rw-rw---- 1 root microk8s        72 Apr 26 12:24 0000000404298038-0000000404298039
-rw-rw---- 1 root microk8s        96 Apr 26 12:32 0000000404298040-0000000404298042
-rw-rw---- 1 root microk8s       144 Apr 26 12:34 0000000404298043-0000000404298047
-rw-rw---- 1 root microk8s        48 Apr 26 12:39 0000000404298048-0000000404298048
-rw-rw---- 1 root microk8s        48 Apr 26 12:40 0000000404298049-0000000404298049
-rw-rw---- 1 root microk8s        48 Apr 26 13:14 0000000404298050-0000000404298050
-rw-rw---- 1 root microk8s       128 Apr 26 13:14 0000000404298051-0000000404298053
-rw-rw---- 1 root microk8s        48 Apr 26 13:58 0000000404298054-0000000404298054
-rw-rw---- 1 root microk8s        88 Apr 26 13:59 0000000404298055-0000000404298056
-rw-rw---- 1 root microk8s        72 Apr 26 14:23 0000000404298057-0000000404298058
-rw-rw---- 1 root microk8s        48 Apr 26 14:40 0000000404298059-0000000404298059
-rw-rw---- 1 root microk8s        88 Apr 26 14:41 0000000404298060-0000000404298061
-rw-rw---- 1 root microk8s       480 Apr 26 14:50 0000000404298062-0000000404298080
-rw-rw---- 1 root microk8s        96 Apr 26 14:50 0000000404298081-0000000404298083
-rw-rw---- 1 root microk8s     16520 Apr 26 14:53 0000000404298084-0000000404298085
-rw-rw---- 1 root microk8s        96 Apr 26 15:44 0000000404298086-0000000404298088
-rw-rw---- 1 root microk8s       120 Apr 26 16:30 0000000404298089-0000000404298092
-rw-rw---- 1 root microk8s        48 Apr 26 16:55 0000000404298093-0000000404298093
-rw-rw---- 1 root microk8s       112 Apr 26 16:56 0000000404298094-0000000404298096
-rw-rw---- 1 root microk8s        72 Apr 26 16:57 0000000404298097-0000000404298098
-rw-rw---- 1 root microk8s        88 Apr 26 16:57 0000000404298099-0000000404298100
-rw-rw---- 1 root microk8s        48 Apr 26 16:58 0000000404298101-0000000404298101
-rw-rw---- 1 root microk8s        72 Apr 26 16:58 0000000404298102-0000000404298103
-rw-rw---- 1 root microk8s        72 Apr 26 18:16 0000000404298104-0000000404298105
-rw-rw---- 1 root microk8s        96 Apr 26 18:31 0000000404298106-0000000404298108
-rw-rw---- 1 root microk8s        48 Apr 26 19:03 0000000404298109-0000000404298109
-rw-rw---- 1 root microk8s        88 Apr 26 19:03 0000000404298110-0000000404298111
-rw-rw---- 1 root microk8s        72 Apr 30 12:48 0000000404298112-0000000404298113
-rw-rw---- 1 root microk8s        48 Apr 30 17:27 0000000404298114-0000000404298114
-rw-rw---- 1 root microk8s        88 Apr 30 17:28 0000000404298115-0000000404298116
-rw-rw---- 1 root microk8s        72 Apr 30 17:28 0000000404298117-0000000404298118
-rw-rw---- 1 root microk8s      1874 Jul 19  2021 cluster.crt
-rw-rw---- 1 root microk8s      3268 Jul 19  2021 cluster.key
-rw-rw---- 1 root microk8s       403 May  1 01:10 cluster.yaml
-rw-rw-r-- 1 root microk8s         2 May  1 01:18 failure-domain
-rw-rw---- 1 root microk8s        62 Jul 19  2021 info.yaml
srw-rw---- 1 root microk8s         0 Apr 26 09:21 kine.sock=
-rw-rw---- 1 root microk8s        68 May  1 01:10 localnode.yaml
-rw-rw---- 1 root microk8s        32 May  1 01:18 metadata1
-rw-rw---- 1 root microk8s        32 May  1 01:18 metadata2
-rw-rw---- 1 root microk8s 117933914 Apr 25 00:44 snapshot-1184-404294957-113597040
-rw-rw---- 1 root microk8s       216 Apr 25 00:44 snapshot-1184-404294957-113597040.meta
-rw-rw---- 1 root microk8s 118216047 Apr 25 00:54 snapshot-1193-404296083-114198360
-rw-rw---- 1 root microk8s       216 Apr 25 00:54 snapshot-1193-404296083-114198360.meta
```
None of the snapshots or all-numeral files are from before the shutdown on any node, unfortunately.

**Comment by balchua** [comment]
@ktsakalozos @neoaggelos sorry to call you out, by any chance you have an idea why kubelite throws segmentation fault?  I think it may have something to do when dqlite starts, at least based on the logs.
The user is in 1.21.11, do you think those dqlite env variables to enable trace logs may help?

**Comment by ktsakalozos** [comment]
Sorry for the late response. Was there any change in the allocated IPs?

It looks like the dqlite cannot form a cluster. Maybe you can recover the cluster as described in https://microk8s.io/docs/restore-quorum but that would mean you would need to remove and rejoin all nodes. Is this an option for you?

**Comment by crookstonmet** [comment]
The IPs didn't change.

Removing and rejoining nodes would be fine. I tried the steps in the 'restore-quorum' link on a couple of the nodes, but none of them would start even when they were the only entry in `cluster.yaml`.

We've decided to rebuild our cluster with microk8s 1.24 and develop better shutdown and backup scripts. I suspect we'd be much better off if we'd run `microk8s stop` on each node in sequence and backed up the contents of `/var/snap/microk8s/current/var/kubernetes/backend/` so we'd have something uncorrupted to restore from.

Thank you both for your attention and help!

**Comment by adubkov** [comment]
@ktsakalozos  Is there any gotchas in how to properly shutdown\start 3+ node cluster? I'm worried not being able to start the cluster normally after power outage. I did not find anything related to it in documentation.

**Comment by ktsakalozos** [comment]
@adubkov, there is nothing special that needs to be done to handle the power outage case. Assuming the nodes will come up on the same IP, the datastore will form a quorum and the cluster will be back online. Please open a separate issue if you see something wrong and we will investigate the case at hand.
